### PR TITLE
refactor(navigation): rework navigation sidebar

### PR DESCRIPTION
**Pull request by MrTimscampi**
Reworks the navigation drawer a bit to look nicer and removes the duplicate user icon (Note: The screenshots were taken before adding the translation key for the header).

Screenshot_2020-11-29 jellyfin-vue Screenshot_2020-11-29 jellyfin-vue(1) Screenshot_2020-11-29 jellyfin-vue(2)

Edit: Added a chevron next to the user name, to indicate the menu
image

### CODE REVIEW

File: components/UserButton.vue [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
-          <v-icon v-else dark>mdi-account</v-icon>
-        </v-avatar>
-      </v-btn>
+      <div class="d-flex" v-bind="attrs" v-on="on">
+        <div class="mr-4">
+          <v-btn icon>
+            <v-avatar>
+              <v-img
+                v-if="userImage"
+                :src="userImage"
```
Comment (Camc314) [inline components/UserButton.vue:10]: Probably better to use the `imageHelper` mixin

**Review by Camc314 — COMMENTED** [review]
LGTM.

On another note, the translation globe now looks a bit lonely on its own in the top bar. I guess this is slightly longer term, but should we use the user settings instead?

**Comment by MrTimscampi** [comment]
> On another note, the translation globe now looks a bit lonely on its own in the top bar. I guess this is slightly longer term, but should we use the user settings instead?

That's the plan. Though, in the long term, the cast and SyncPlay icons are going to live there as well.

**Comment by Extarys** [comment]
When using a freshly opened window, the dark mode toggle is defaulted to false, even though the dark mode is on.
image

Could this be fixed in this PR?

File: components/UserButton.vue (hunk 1 above)
Comment (MrTimscampi) [reply components/UserButton.vue:10]: The main issue (for now) is that the current imageHelper mixing handles items (and a user isn't an item :/ )

I want to refactor it to clean it up and make it more modular, eventually, but I think that's outside the scope of this PR.

Comment (Camc314) [reply components/UserButton.vue:10]: Ok, sounds good.

File: components/UserButton.vue [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
-            :src="userImage"
-            :alt="$auth.user.Name"
-          ></v-img>
-          <v-icon v-else dark>mdi-account</v-icon>
-        </v-avatar>
-      </v-btn>
+      <div class="d-flex" v-bind="attrs" v-on="on">
+        <div class="mr-4">
+          <v-btn icon>
+            <v-avatar>
```
Comment (Camc314) [inline components/UserButton.vue:7]: Not essential, but I think it looks slightly better with a background color on the avatar icon

**Comment by Extarys** [comment]
Tested it, looks neat. I love it.

**Comment by dkanada** [comment]
What did the previous interface look like?

**Comment by Camc314** [comment]
image
image
image

Something like this

File: components/UserButton.vue (hunk 2 above)
Comment (MrTimscampi) [reply components/UserButton.vue]: Indeed, I think it was removed when moving stuff around between the two user picture components to make a single one. I didn't notice it due to having an avatar set at the time.

It's fixed in 53015bd :sweat_smile:

**Review by Camc314 — APPROVED** [review]

**Pull request merged**